File: frontend/lib/core/models/receta.dart
class Receta {
  final int id;
  final String titulo;
  final String? descripcion;
  final int? tiempoPreparacionMinutos;
  final String? dificultad;
  final String? imagenUrl;
  final bool esTip;
  final Set<String> etiquetas;
  final List<String> ingredientes;
  final List<String> pasos;

  Receta({
    required this.id,
    required this.titulo,
    this.descripcion,
    this.tiempoPreparacionMinutos,
    this.dificultad,
    this.imagenUrl,
    required this.esTip,
    required this.etiquetas,
    required this.ingredientes,
    required this.pasos,
  });

  factory Receta.fromJson(Map<String, dynamic> json) => Receta(
        id: json['id'],
        titulo: json['titulo'] ?? '',
        descripcion: json['descripcion'],
        tiempoPreparacionMinutos: json['tiempoPreparacionMinutos'],
        dificultad: json['dificultad'],
        imagenUrl: json['imagenUrl'],
        esTip: json['esTip'] ?? false,
        etiquetas: ((json['etiquetas'] as List?) ?? []).map((e) => e.toString()).toSet(),
        ingredientes: ((json['ingredientes'] as List?) ?? []).map((e) => e.toString()).toList(),
        pasos: ((json['pasos'] as List?) ?? []).map((e) => e.toString()).toList(),
      );

  Map<String, dynamic> toJson() => {
        'titulo': titulo,
        'descripcion': descripcion,
        'tiempoPreparacionMinutos': tiempoPreparacionMinutos,
        'dificultad': dificultad,
        'imagenUrl': imagenUrl,
        'esTip': esTip,
        'etiquetas': etiquetas.toList(),
        'ingredientes': ingredientes,
        'pasos': pasos,
      };
}

File: backend/src/main/java/com/safebite/backend/model/Receta.java
package com.safebite.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "recetas")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Receta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titulo;

    @Column(length = 2000)
    private String descripcion;

    private Integer tiempoPreparacionMinutos;

    private String dificultad;

    private String imagenUrl;

    private boolean esTip;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "receta_etiquetas", joinColumns = @JoinColumn(name = "receta_id"))
    @Column(name = "etiqueta")
    @Builder.Default
    private Set<String> etiquetas = new HashSet<>();

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "receta_ingredientes", joinColumns = @JoinColumn(name = "receta_id"))
    @Column(name = "ingrediente", length = 500)
    @OrderColumn(name = "orden")
    @Builder.Default
    private List<String> ingredientes = new ArrayList<>();

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "receta_pasos", joinColumns = @JoinColumn(name = "receta_id"))
    @Column(name = "paso", length = 1000)
    @OrderColumn(name = "orden")
    @Builder.Default
    private List<String> pasos = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "publicado_por")
    private Usuario publicadoPor;

    @Builder.Default
    private LocalDateTime fechaPublicacion = LocalDateTime.now();
}
